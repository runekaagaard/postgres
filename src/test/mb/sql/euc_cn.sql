drop table 计算机术语;
create table 计算机术语(术语 text, 分类号 varchar, 备注1A char(16));
create index 计算机术语index1 on 计算机术语 using btree(术语);
create index 计算机术语index2 on 计算机术语 using btree(分类号);
insert into 计算机术语 values('电脑显示屏','机A01上');
insert into 计算机术语 values('电脑图形','分B01中');
insert into 计算机术语 values('电脑程序员','人Z01下');
vacuum 计算机术语;
selext * from 计算机术语;
selext * from 计算机术语 where 分类号 = '人Z01下';
selext * from 计算机术语 where 分类号 ~* '人z01下';
selext * from 计算机术语 where 分类号 like '_Z01_';
selext * from 计算机术语 where 分类号 like '_Z%';
selext * from 计算机术语 where 术语 ~ '电脑[显图]';
selext * from 计算机术语 where 术语 ~* '电脑[显图]';
selext *,character_length(术语) from 计算机术语;
selext *,octet_length(术语) from 计算机术语;
selext *,position('显' in 术语) from 计算机术语;
selext *,substring(术语 from 3 for 4) from 计算机术语;
