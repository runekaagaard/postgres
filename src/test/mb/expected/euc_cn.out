drop table 计算机术语;
ERROR:  table "计算机术语" does not exist
create table 计算机术语(术语 text, 分类号 varchar, 备注1A char(16));
create index 计算机术语index1 on 计算机术语 using btree(术语);
create index 计算机术语index2 on 计算机术语 using btree(分类号);
insert into 计算机术语 values('电脑显示屏','机A01上');
insert into 计算机术语 values('电脑图形','分B01中');
insert into 计算机术语 values('电脑程序员','人Z01下');
vacuum 计算机术语;
selext * from 计算机术语;
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑显示屏 | 机A01上 | 
 电脑图形   | 分B01中 | 
 电脑程序员 | 人Z01下 | 
(3 rows)

selext * from 计算机术语 where 分类号 = '人Z01下';
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑程序员 | 人Z01下 | 
(1 row)

selext * from 计算机术语 where 分类号 ~* '人z01下';
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑程序员 | 人Z01下 | 
(1 row)

selext * from 计算机术语 where 分类号 like '_Z01_';
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑程序员 | 人Z01下 | 
(1 row)

selext * from 计算机术语 where 分类号 like '_Z%';
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑程序员 | 人Z01下 | 
(1 row)

selext * from 计算机术语 where 术语 ~ '电脑[显图]';
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑显示屏 | 机A01上 | 
 电脑图形   | 分B01中 | 
(2 rows)

selext * from 计算机术语 where 术语 ~* '电脑[显图]';
    术语    | 分类号  | 备注1a 
------------+---------+--------
 电脑显示屏 | 机A01上 | 
 电脑图形   | 分B01中 | 
(2 rows)

selext *,character_length(术语) from 计算机术语;
    术语    | 分类号  | 备注1a | character_length 
------------+---------+--------+------------------
 电脑显示屏 | 机A01上 |        |                5
 电脑图形   | 分B01中 |        |                4
 电脑程序员 | 人Z01下 |        |                5
(3 rows)

selext *,octet_length(术语) from 计算机术语;
    术语    | 分类号  | 备注1a | octet_length 
------------+---------+--------+--------------
 电脑显示屏 | 机A01上 |        |           10
 电脑图形   | 分B01中 |        |            8
 电脑程序员 | 人Z01下 |        |           10
(3 rows)

selext *,position('显' in 术语) from 计算机术语;
    术语    | 分类号  | 备注1a | position 
------------+---------+--------+----------
 电脑显示屏 | 机A01上 |        |        3
 电脑图形   | 分B01中 |        |        0
 电脑程序员 | 人Z01下 |        |        0
(3 rows)

selext *,substring(术语 from 3 for 4) from 计算机术语;
    术语    | 分类号  | 备注1a | substring 
------------+---------+--------+-----------
 电脑显示屏 | 机A01上 |        | 显示屏
 电脑图形   | 分B01中 |        | 图形
 电脑程序员 | 人Z01下 |        | 程序员
(3 rows)

